### PR TITLE
feat: allow external services to be authable

### PR DESCRIPTION
**Pull request by kylecarbs**
Allows external services to be authenticated with Coder via "Create Workspace".

Shortlist of changes:
1. Renamed environment variables from `CODER_GITAUTH_` to `CODER_EXTERNAL_AUTH_` (this is backwards compatible).
2. New `DISPLAY_NAME` and `DISPLAY_ICON` properties.
3. Newew `TYPE=oidc` to enable *any* OpenID Connect provider to work with external auth.

image

Future improvements:
1. Add a sub-command to the Coder CLI to dynamically obtain the access token from a provider. e.g. `coder external-auth refresh --type=github`, `coder external-auth refresh --id=github`, `coder external-auth refresh --match="https://github.com"`
2. Add `coder_external_auth` Terraform data source to replace `coder_git_auth`. Allow for matching based on `type` and `regex` so Modules can make use of auth 👀.

### CODE REVIEW

**Comment by kylecarbs** [comment]
@bpmct please review the docs on this change.

We also need to add a permanent redirect for coder.com from `/docs/v2/admin/git-auth` to `/docs/v2/admin/external-auth`.

File: site/src/pages/ExternalAuthPage/ExternalAuthPage.tsx [hunk 1]
```diff
@@ -72,7 +72,7 @@ const ExternalAuthPage: FC = () => {
     !getExternalAuthProviderQuery.data.authenticated &&
     !getExternalAuthProviderQuery.data.device
   ) {
-    window.location.href = `/externalauth/${provider}/callback`;
+    window.location.href = `/external-auth/${provider}/callback`;
```
Comment (BrunoQuaresma) [inline site/src/pages/ExternalAuthPage/ExternalAuthPage.tsx:75]: Would be nice to use `navigate` instead

File: site/src/pages/ExternalAuthPage/ExternalAuthPageView.tsx [hunk 2]
```diff
@@ -51,7 +51,7 @@ const ExternalAuthPageView: FC<ExternalAuthPageViewProps> = ({
 
   // We only want to wrap this with a link if an install URL is available!
   let installTheApp: JSX.Element = (
-    <>{`install the ${externalAuth.type} App`}</>
+    <>{`install the ${externalAuth.display_name} App`}</>
```
Comment (BrunoQuaresma) [inline site/src/pages/ExternalAuthPage/ExternalAuthPageView.tsx:54]: You can change the type to be `ReactNode` and use a string directly instead

**Review by BrunoQuaresma — APPROVED** [review]
FE code looks good but there are a few Storybook stories broken

**Comment by kylecarbs** [comment]
Fixed! Good call on checking the Storybook.

**Pull request merged**